reactor-netty-core/src/main/java/reactor/netty/channel/ChannelMeters.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2022 VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.channel;
+
+import io.micrometer.api.instrument.Meter;
+import io.micrometer.api.instrument.docs.DocumentedMeter;
+import io.micrometer.api.instrument.docs.TagKey;
+
+/**
+ * Channel meters.
+ *
+ * @author Marcin Grzejszczak
+ * @author Violeta Georgieva
+ * @since 1.1.0
+ */
+public enum ChannelMeters implements DocumentedMeter {
+
+	/**
+	 * The number of all opened connections on the server.
+	 */
+	CONNECTIONS_TOTAL {
+		@Override
+		public String getName() {
+			return "%s";
+		}
+
+		@Override
+		public TagKey[] getTagKeys() {
+			return ConnectionsTotalMeterTags.values();
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.GAUGE;
+		}
+	},
+
+	/**
+	 * Amount of the data received, in bytes.
+	 */
+	DATA_RECEIVED {
+		@Override
+		public String getBaseUnit() {
+			return "bytes";
+		}
+
+		@Override
+		public String getName() {
+			return "%s";
```

Comment:
can you clarify the `%s`?
from what I understand:
 -  the name of the metric is either explicitly set in the enum (like in `reactor.netty.connection.provider.active.connections`)
 - OR the name of the metric is set to `"%s"` and somehow it will resolve what's configured as name in the builder? (eg in `MicrometerChannelMetricsRecorder`, still uses the `Metrics.DATA_RECEIVED` suffix + a variable `name` for prefix, resulting in eg. `reactor.netty.tcp.client.data.received`)
 - the link to the generated adoc section uses the enum's name? (`DATA_RECEIVED` => `<<observability-metrics-data-received>>`, `ERROR_COUNT` => `<<observability-metrics-error-count>>`, `ACTIVE_CONNECTIONS` => `observability-metrics-active-connections`...)

Reply:
When we know in advance the name we can provide it directly in the `DocumentedMeter`, for example this is the case with `reactor.netty.connection.provider.active.connections`. Then in the generated documentation we have

```
**Metric name** `reactor.netty.connection.provider.active.connections`. **Type** `gauge`.
```

When the name is dynamic in the `DocumentedMeter` we specify `%s`, then the generated documentation looks like this

```
**Metric name** `%s` - since it contains `%s`, the name is dynamic and will be resolved at runtime. **Type** `gauge`.
```

such metrics are 
`reactor.netty.tcp.client.data.received`
`reactor.netty.tcp.server.data.received`
`reactor.netty.http.client.data.received`
`reactor.netty.http.server.data.received`

Reactor Netty provides `data.reveived` metric for every server/client type.